commands/init.go

```diff
@@ -4,58 +4,80 @@
 package commands
 
 import (
+	"fmt"
 	"strings"
 
 	"github.com/mattermost/mattermost-server/v5/model"
 	"github.com/mattermost/mmctl/client"
+	"github.com/mattermost/mmctl/printer"
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 )
 
+func CheckVersionMatch(version, serverVersion string) bool {
+	maj, min, _ := model.SplitVersion(version)
+	srvMaj, srvMin, _ := model.SplitVersion(serverVersion)
+
+	return maj == srvMaj && min == srvMin
+}
+
 func withClient(fn func(c client.Client, cmd *cobra.Command, args []string) error) func(cmd *cobra.Command, args []string) error {
 	return func(cmd *cobra.Command, args []string) error {
-		c, err := InitClient()
+		c, serverVersion, err := InitClient()
 		if err != nil {
 			return err
 		}
+
+		valid := CheckVersionMatch(Version, serverVersion)
+		if !valid {
+			if viper.GetBool("strict") {
```

Comment:
One question to understand better. Do we need viper as a new dependency?

I mean, we can just create a new method in `version.go` called something like `SetStrict` like we have with the `printer.SetFormat` and then use it here as we used `Version` variable in the line 32

Reply:
I'm not sure I'm understanding your point here. Viper is used so when reading the `strict` value, it takes into consideration both the `--strict` flag and the `MMCTL_STRICT` environment variable. As a side benefit, I've binded as well the `MMCTL_FORMAT` environment variable, so now both can be received as flags or env variables.

The rationale behind this is that we want to allow the admins using `mmctl` that want all their commands to be strict to be able to do it easily, and an environment variable is a great mechanism for this.

Reply:
Cool, understood! Thank you